### PR TITLE
Convert smoke.e2e.tests and DiscoveryAndExecutionTests to use local test framework

### PR DESCRIPTION
**Pull request by engyebrahim**

### CODE REVIEW

File: test/E2ETests/Smoke.E2E.Tests/TestDataSourceTests.cs [hunk 1]
```diff
@@ -4,17 +4,14 @@
...
 using Microsoft.MSTestV2.CLIAutomation;
-using Microsoft.VisualStudio.TestTools.UnitTesting;
 
-[TestClass]
 public class TestDataSourceTests : CLITestBase
 {
     private const string TestAssembly = "DataSourceTestProject.dll";
 
+    // Made it private because it Was Ignored.
     // TODO @haplois | @evangelink: This test fails under CI - will be fixed in a future PR
```
Comment (Evangelink) [inline test/E2ETests/Smoke.E2E.Tests/TestDataSourceTests.cs]:
```suggestion
    // TODO: @haplois | @evangelink: This test fails under CI - will be fixed in a future PR (marked as private to ignore the test)
```

File: test/TestFramework.ForTestingMSTest/AdapterToTestPlatform.cs [hunk 2]
```diff
@@ -128,6 +128,19 @@ private static void LogMessage(IMessageLogger? logger, TestMessageLevel level, s
         }
     }
 
+    // Looking up base types.
+    private static bool IsAnyBaseClassInheritingTestContainer(Type typeInfo)
```
Comment (Evangelink) [inline test/TestFramework.ForTestingMSTest/AdapterToTestPlatform.cs]: I would recommend renaming this into either `IsInheritingTestContainer` or `IsTestContainer`.

File: test/TestFramework.ForTestingMSTest/AdapterToTestPlatform.cs [hunk 3]
```diff
@@ -128,6 +128,19 @@ private static void LogMessage(IMessageLogger? logger, TestMessageLevel level, s
...
+    {
+        while(typeInfo != null)
```
Comment (Evangelink) [inline test/TestFramework.ForTestingMSTest/AdapterToTestPlatform.cs]: Nit: 
```suggestion
        while (typeInfo != null)
```

File: test/TestFramework.ForTestingMSTest/AdapterToTestPlatform.cs [hunk 4]
```diff
@@ -128,6 +128,19 @@ private static void LogMessage(IMessageLogger? logger, TestMessageLevel level, s
...
+        {
+            if(typeInfo == typeof(TestContainer))
```
Comment (Evangelink) [inline test/TestFramework.ForTestingMSTest/AdapterToTestPlatform.cs]: Nit:

```suggestion
            if (typeInfo == typeof(TestContainer))
```

**Review by Evangelink — DISMISSED** [review]
A few nitpicks but overall LGTM! Thank you for taking the time to update the method finding test containers 👍

**Review by Evangelink — APPROVED** [review]

**Review by Evangelink — APPROVED** [review]

File: test/E2ETests/DiscoveryAndExecutionTests/DiscoveryAndExecutionTests.csproj [hunk 5]
```diff
@@ -19,6 +19,10 @@
     <PackageReference Include="System.ValueTuple" Version="$(SystemValueTupleVersion)" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="$(RepoRoot)test\TestFramework.ForTestingMSTest\TestFramework.ForTestingMSTest.csproj" />
+  </ItemGroup>
+
   <ItemGroup>
     <Reference Include="Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter">
```
Comment (Evangelink) [inline test/E2ETests/DiscoveryAndExecutionTests/DiscoveryAndExecutionTests.csproj]: Could you make a test to see if we need to manually reference all these dlls? (Make a run where you try to remove them).

File: test/E2ETests/DiscoveryAndExecutionTests/Extensions/AssertionExtensions.cs [hunk 6]
```diff
@@ -5,54 +5,55 @@ namespace Microsoft.MSTestV2.Smoke.DiscoveryAndExecutionTests;
...
+using TestFramework.ForTestingMSTest;
 
 using System.Collections.Generic;
 using System.Linq;
 
-using Assert = Microsoft.VisualStudio.TestTools.UnitTesting.Assert;
+using TestContainer = TestFramework.ForTestingMSTest.TestContainer;
 
-public static class AssertionExtensions
+public static class AssertionExtensions
```
Comment (Evangelink) [inline test/E2ETests/DiscoveryAndExecutionTests/Extensions/AssertionExtensions.cs]: I would rename this into something like `VerifyE2E`.

File: test/E2ETests/DiscoveryAndExecutionTests/DiscoveryAndExecutionTests.csproj (hunk 5 above)
Comment (engyebrahim) [reply test/E2ETests/DiscoveryAndExecutionTests/DiscoveryAndExecutionTests.csproj]: if I remove it, I get compilation errors

**Review by Evangelink — DISMISSED** [review]

File: test/E2ETests/DiscoveryAndExecutionTests/DiscoveryAndExecutionTests.csproj (hunk 5 above)
Comment (Evangelink) [reply test/E2ETests/DiscoveryAndExecutionTests/DiscoveryAndExecutionTests.csproj]: Ok so let's keep things as-is. Thanks for the confirmation.

File: test/E2ETests/DiscoveryAndExecutionTests/TestDataSourceTests.cs [hunk 7]
```diff
@@ -4,20 +4,14 @@
...
-[TestClass]
 public class TestDataSourceTests : CLITestBase
 {
     private const string TestAssembly = "DataSourceTestProject.dll";
 
-    // TODO @haplois | @evangelink: This test fails under CI - will be fixed in a future PR
-    [Ignore]
-    [TestMethod]
+    // TODO @haplois | @evangelink: This test fails under CI - will be fixed in a future PR (Marked as private to ignore the test)
     public void ExecuteCsvTestDataSourceTests()
```
Comment (Evangelink) [inline test/E2ETests/DiscoveryAndExecutionTests/TestDataSourceTests.cs]: @engyebrahim This is your issue :) Test was ignored and is no longer.

**Pull request merged**